packages/destination-subscriptions/src/parse-fql.ts

```diff
@@ -1,402 +1,439 @@
-					if (isExists) {
-						nodes.push({
-							type: 'event-trait',
-							name: token.value.replace(/^(traits)\./, ''),
-							operator: 'exists'
-						})
-					} else if (isNotExists) {
-						nodes.push({
-							type: 'event-trait',
-							name: token.value.replace(/^(traits)\./, ''),
-							operator: 'not_exists'
-						})
-					} else {
-						nodes.push({
-							type: 'event-trait',
-							name: token.value.replace(/^(traits)\./, ''),
-							operator: operatorToken.value as Operator,
-							value: getTokenValue(valueToken)
-						})
-					}
-				} else if (conditionType === 'event-context') {
-					if (isExists) {
-						nodes.push({
-							type: 'event-context',
-							name: token.value.replace(/^(context)\./, ''),
-							operator: 'exists'
-						})
-					} else if (isNotExists) {
-						nodes.push({
-							type: 'event-context',
-							name: token.value.replace(/^(context)\./, ''),
-							operator: 'not_exists'
-						})
-					} else {
-						nodes.push({
-							type: 'event-context',
-							name: token.value.replace(/^(context)\./, ''),
-							operator: operatorToken.value as Operator,
-							value: getTokenValue(valueToken)
-						})
-					}
-				}
-			}
-
-			if (isFqlFunction(token)) {
-				parseFqlFunction(token.value, nodes, tokens)
-			}
-		}
-
-		if (token.type === 'operator' && token.value === '!') {
-			if (isFqlFunction(tokens[0])) {
-				const name = tokens[0].value
-				tokens.shift()
-				parseFqlFunction(name, nodes, tokens, { negate: true })
-			}
-		}
-
-		if (token.type === 'parenleft') {
-			const groupTokens: Token[] = []
-			let groupToken = tokens.shift()
-
-			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-			while (groupToken!.type !== 'parenright') {
-				// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-				groupTokens.push(groupToken!)
-				groupToken = tokens.shift()
-			}
-
-			groupTokens.push({ type: TokenType.EOS, value: 'eos' })
-			nodes.push(parse(groupTokens))
-		}
-
-		if (token.type === 'conditional') {
-			operator = token.value
-		}
-
-		token = tokens.shift()
-	}
-
-	if (nodes.length > 1) {
-		return {
-			type: 'group',
-			operator: operator as GroupConditionOperator,
-			children: nodes
-		}
-	}
-
-	return nodes[0]
+  const nodes: Condition[] = []
+  let operator = 'and'
+  let token = tokens.shift()
+
+  while (token && token.type !== 'eos') {
+    if (token.type === 'ident') {
+      const [tokenStart] = (token.value ?? '').split('.')
+      const conditionType = tokenToConditionType[tokenStart]
+
+      if (conditionType) {
+        const operatorToken = tokens.shift()
+
+        if (!operatorToken) {
+          throw new Error('Operator token is missing')
+        }
+
+        const valueToken = tokens.shift()
+
+        if (!valueToken) {
+          throw new Error('Value token is missing')
+        }
+
+        const isTrue = operatorToken.value === '=' && checkBooleanTrue(valueToken.value) === true
```

Comment:
`checkBooleanTrue` feels a little too heavy. is this equivalent?

```suggestion
        const isTrue = operatorToken.value === '=' && valueToken.value === 'true'
```